code/modules/vehicles/mecha/mecha_defense.dm

```diff
@@ -41,7 +41,7 @@
 	if(damage_taken <= 0 || atom_integrity < 0)
 		return damage_taken
 
-	spark_system.start()
+	spark_system?.start()
```

Comment:
``spark_system`` is new'ed in the Initialize and is only deleted on Destroy, why are we assuming it might not exist?

Reply:
I... said in the PR body????????????


ln 40.
/obj/vehicle/sealed/mecha/take_damage()
	..() = /atom/proc/take_damage()
	
/atom/proc/take_damage() -> atom_destruction()
/obj/vehicle/sealed/mecha/atom_destruction()
	..() = /obj/atom_destruction()

/obj/atom_destruction() -> /obj/deconstruct()
/obj/deconstruct() -> qdel(src)
qel(src) -> /obj/vehicle/sealed/mecha/Destroy()
/obj/vehicle/sealed/mecha/Destroy() -> QDEL_NULL(spark_system)

NOW spark_system.start() is called.